src/Npgsql/TypeHandlers/LQueryHandler.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Npgsql.BackendMessages;
+using Npgsql.PostgresTypes;
+using Npgsql.TypeHandling;
+using Npgsql.TypeMapping;
+using NpgsqlTypes;
+
+namespace Npgsql.TypeHandlers
+{
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+    [TypeMapping("lquery", NpgsqlDbType.LQuery)]
+    class LQueryHandlerFactory : NpgsqlTypeHandlerFactory<string>
+    {
+        public override NpgsqlTypeHandler<string> Create(PostgresType postgresType, NpgsqlConnection conn)
+            => new LQueryHandler(postgresType, conn);
+    }
+
+    /// <summary>
+    /// LQuery binary encoding is a simple UTF8 string, but prepended with a version number.
+    /// </summary>
+    public class LQueryHandler : TextHandler
+    {
+        /// <summary>
+        /// Prepended to the string in the wire encoding
+        /// </summary>
+        const byte LQueryProtocolVersion = 1;
+
+        internal override bool PreferTextWrite => false;
+
+        protected internal LQueryHandler(PostgresType postgresType, NpgsqlConnection connection)
+            : base(postgresType, connection) {}
+
+        #region Write
+
+        public override int ValidateAndGetLength(string value, ref NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter)
+        {
+            if (lengthCache == null)
+                lengthCache = new NpgsqlLengthCache(1);
+            if (lengthCache.IsPopulated)
+                return lengthCache.Get() + 1;
```

Comment:
Not sure at what time I wrote this (and same issue in the other handlers) but it seems plain wrong :)

Reply:
It's not wrong because the value is populated by the base class, so the version byte isn't included.